iothub/device/src/Transport/AmqpIot/AmqpIotExceptionAdapter.cs

```diff
@@ -40,9 +40,7 @@ internal static Exception ConvertToIotHubException(Exception exception)
         internal static Exception ConvertToIotHubException(Exception exception, AmqpObject source)
         {
             Exception ex = ConvertToIotHubException(exception);
-            if (source.IsClosing() &&
-                (ex is InvalidOperationException
-                || ex is OperationCanceledException))
+            if (source.IsClosing() && ex is InvalidOperationException)
             {
                 return new IotHubClientException("Amqp resource is disconnected.", null, IotHubStatusCode.NetworkErrors);
```

Comment:
Upper case AMQP, and also we should set the inner exception.
```suggestion
                return new IotHubClientException("AMQP resource is disconnected.", null, IotHubStatusCode.NetworkErrors);
```